core/commonMain/src/kotlinx/serialization/Serializers.kt

```diff
@@ -108,6 +126,32 @@ public fun SerializersModule.serializer(type: KType): KSerializer<Any?> =
     serializerByKTypeImpl(type, failOnMissingTypeArgSerializer = true) ?: type.kclass()
         .platformSpecificSerializerNotRegistered()
 
+
+/**
+ * Retrieves serializer for the given [type] and,
+ * if [type] is not serializable, fallbacks to [contextual][SerializersModule.getContextual] lookup.
+ *
+ * **Retrieving an array serializer is not supported**
+ *
+ * If [type] is a parametrized type then in [typeArgumentsSerializers] it is necessary to pass serializers for generic parameters.
+ *
+ * The nullability of returned serializer is specified using the [isNullable].
+ *
+ * Caching on JVM platform is disabled for this function, so it may work slower than overloading with [KType].
+ *
+ * @throws SerializationException if serializer cannot be created (provided [type] or its type argument is not serializable and is not registered in [this] module).
+ * @throws SerializationException if [type] is an array class
+ * @throws Exception if size of [typeArgumentsSerializers] does not match the expected generic parameters count
+ */
+@ExperimentalSerializationApi
+public fun <T : Any> SerializersModule.serializer(
+    type: KClass<T>,
+    typeArgumentsSerializers: List<KSerializer<*>>,
+    isNullable: Boolean
+): KSerializer<Any?> =
```

Comment:
By the way, why return type is `KSerializer<Any>` and not `KSerializer<T>`?

Reply:
Made only because of the uniformity with the functions above.

At the same time, I don't think it will be of much use, because unlike `typeOf<...>()`, `*` will be used as parameters and for `serializer(Parametrized::class, listOf(Int.serializer()), false)` it will still be uncomfortable.

Reply:
KClass in overload without module should still have T parameter imho, for consistency

Reply:
I think it makes sense to leave return type as `KSerializer<Any>` because T may not have information about `isNullable` and `typeParametersSerializers`